index.js

```diff
@@ -6,6 +6,60 @@ const KEYCODE = {
 }
 
 const state = new Map()
+// when container or children get focus
+const onFocusin = e => {
+  const {currentTarget: rover} = e
+  if (state.get('last_rover') == rover) return
+  if (state.has(rover)) {
+    activate(rover, state.get(rover).active)
+    state.set('last_rover', rover)
+  }
+}
+const onKeydown = e => {
+  const {currentTarget: rover} = e
+  switch (e.keyCode) {
+    case KEYCODE[isRtl ? 'LEFT' : 'RIGHT']:
+    case KEYCODE.DOWN:
+      e.preventDefault()
+      focusNextItem(rover)
+      break
+    case KEYCODE[isRtl ? 'RIGHT' : 'LEFT']:
+    case KEYCODE.UP:
+      e.preventDefault()
+      focusPreviousItem(rover)
+      break
+  }
+}
+const mo = new MutationObserver((mutationList, observer) =>{
+  const stateElementsSet = new Set();
+  state.forEach((v,k) =>{
+    stateElementsSet.add(k.classList[0])
+  } )
+  const qs = [...stateElementsSet].map(x => `.${x}`).join(',');
+  mutationList.forEach(mutation => {
+     if(mutation.removedNodes.length > 0){
+       mutation.removedNodes.forEach(removedEl => {
+         if (removedEl.nodeType !== 1) return //only elements
+         const e = state.has(removedEl) 
+         ? removedEl 
+         : removedEl.querySelector(qs); 
```

Comment:
instead of trying to stash query selectors into stateElementsSet (which i imagine wont always be unique and not always insert into the set), how about looping over the state map and seeing if e.contains (https://developer.mozilla.org/en-US/docs/Web/API/Node/contains) any of the roving nodes we're managing? if e.contains() comes back true with a node, you'll know which (and maybe it's more than 1??) to destroy from our internal state.

Reply:
Well the set would be classes from the state elements but there is a  problem if the rover element was chosen with a non-class selector and the element didn't have a class on it which occurred to me later. Trying contains now.